nucypher/blockchain/eth/actors.py

```diff
@@ -260,9 +260,23 @@ def connect_condition_providers(
             providers = set()
             for uri in condition_blockchain_endpoints:
                 provider = self._make_condition_provider(uri)
+                if int(Web3(provider).eth.chain_id) != int(chain_id):
+                    raise self.ActorError(
+                        f"Condition blockchain endpoint {uri} is not on chain {chain_id}"
+                    )
```

Comment:
Not for this PR, but maybe this could be generalized to be checked even more eagerly, probably at `UrsulaConfiguration` time.